subsystem/kubeapi_test.go

```diff
@@ -120,34 +121,50 @@ func waitForClusterCRDState(
 		}
 		time.Sleep(time.Second)
 	}
-	Expect(cluster.Status.State).Should(Equal(state))
+	Expect(clusterDeployment.Status.Conditions[0].Message).Should(Equal(state))
 	successInARaw++
 	Expect(successInARaw).Should(Equal(minSuccessesInRow))
 }
 
-func getDefaultClusterSpec(secretRef *corev1.SecretReference) *v1alpha1.ClusterSpec {
-	return &v1alpha1.ClusterSpec{
-		Name:                     "test-cluster",
-		OpenshiftVersion:         "4.6",
-		BaseDNSDomain:            "test.domain",
-		ClusterNetworkCidr:       "10.128.0.0/14",
-		ClusterNetworkHostPrefix: 23,
-		ServiceNetworkCidr:       "172.30.0.0/16",
-		IngressVip:               "1.2.3.9",
-		APIVip:                   "1.2.3.8",
-		SSHPublicKey:             sshPublicKey,
-		VIPDhcpAllocation:        false,
-		PullSecretRef:            secretRef,
-		ProvisionRequirements: v1alpha1.ProvisionRequirements{
-			ControlPlaneAgents: 3,
-			WorkerAgents:       0,
-			AgentSelector:      metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+func getDefaultClusterDeploymentSpec(secretRef *corev1.LocalObjectReference) *hivev1.ClusterDeploymentSpec {
+	return &hivev1.ClusterDeploymentSpec{
+		ClusterName: "test-cluster",
+		BaseDomain:  "hive.example.com",
+		Provisioning: &hivev1.Provisioning{
+			InstallConfigSecretRef: &corev1.LocalObjectReference{Name: "cluster-install-config"},
+			ImageSetRef:            &hivev1.ClusterImageSetReference{Name: "openshift-v4.7.0"},
+			InstallStrategy: &hivev1.InstallStrategy{
+				Agent: &agentv1.InstallStrategy{
+					Networking: agentv1.Networking{
+						MachineNetwork: []agentv1.MachineNetworkEntry{},
+						ClusterNetwork: []agentv1.ClusterNetworkEntry{{
+							CIDR:       "10.128.0.0/14",
+							HostPrefix: 23,
+						}},
+						ServiceNetwork: []string{"172.30.0.0/16"},
+					},
+					SSHPublicKey: sshPublicKey,
+					ProvisionRequirements: agentv1.ProvisionRequirements{
+						ControlPlaneAgents: 3,
+						WorkerAgents:       0,
+					},
+				},
+			},
 		},
+		Platform: hivev1.Platform{
+			AgentBareMetal: &agentv1.BareMetalPlatform{
+				APIVIP:            "1.2.3.8",
+				APIVIPDNSName:     "test.domain",
```

Comment:
Need to remove this, it's relevant only for day2 clusters. 
This is what makes your test fail.Update cluster ignore it so `updateIfNeeded` always see a difference and try to update it again, there is no error in the logs for it.